Support larger EDH keys and small cleanups.

nbougalis opened the pull request:
In the past we used only 512-bit EDH keys. To protect against the LOGJAM attack, OpenSSL 1.0.1n and later versions reject EDH keys that are less than 768 bits and a future version will increase the minimum to 1024.

We now have DH parameters for 1024-bit and 2048-bit EDH keys and will correctly return and use those, when asked. Note that at this time, OpenSSL will only ask for 512 or 1024 bit keys.

The reverted commit was a workaround for client-side functionality added to IE to protect against LOGJAM. By properly supporting larger keys we no longer need it.

JoelKatz commented:
:+1:

JoelKatz commented:
I'm not quite sure why the NodeIdentify commit moves the raw DH params code. Was that intentional?

nbougalis commented:
Yes: the function is no longer needed and doesn't need to be exposed public, so I removed the public header and moved it inside the private namespace, just before its use.
